### DISCUSSION TITLE
Disable certificate check when initializing a project?

### DISCUSSION THREAD

**Discussion by iiicebearrr** [reflex-dev/reflex#2081] [General]
When i try to initialize a project, got the certificate error:

```
─────────────────────────────────────────────────────── Initializing app ───────────────────────────────────────────────────────
Debug: Downloading https://github.com/Schniz/fnm/releases/download/v1.35.1/fnm-linux.zip
Debug: Skipping bun installation as it is already installed.
An error occurred while downloading fnm package: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local
issuer certificate (_ssl.c:1000)
```

Will it better to disable the certificate check or make it an optional choice from cli? 

For example, add `verify=False` to:
```python3
def download_and_extract_fnm_zip():
    ...
    try:
        # Download the FNM zip release.
        # TODO: show progress to improve UX
        with httpx.stream("GET", url, follow_redirects=True) as response:
    ...
```

```python3
def download_and_extract_fnm_zip():
    ...
    try:
        # Download the FNM zip release.
        # TODO: show progress to improve UX
        with httpx.stream("GET", url, follow_redirects=True, verify=False) as response:
    ...
```

Or, from cli:
`reflex init --disable-cert-check`

I initialize this project with docker, and the `Dockerfile`:
```dockerfile
FROM python:3.12

WORKDIR /app

RUN pip install poetry

RUN apt-get update -y && apt-get install -y fish git

RUN chsh -s /usr/bin/fish

EXPOSE 3333

CMD ["tail", "-f", "/dev/null"]
```

**Comment by Jogenist** [comment]
I am having exactly the same issue when initializing a project with docker within a company network!
From my point of view, it would be nice if the certificate check would be optional.

**Comment by masenf** [comment]
In the short term, is it possible to use `SSL_CERT_FILE` to specify path to MiTM cert bundle that your company is using? I'm hesitant to plumb through cert disabling option because of the security implications, but will discuss with the team.

**Reply by Xinja-T** [reply]
Is there any update on this issue? 
I attempted to resolve it by assigning the combination of my organization's `REQUESTS_CA_BUNDLE `and the output of `$(python -m certifi)` to `SSL_CERT_FILE`. However, the curl command failed, exiting with code 56. Additionally, unable to fetch app templates due to an `[SSL: CERTIFICATE_VERIFY_FAILED]` error. 
Could you please provide guidance on how to fix this problem?

**Comment by rahulswarnkar** [comment]
I use `REQUESTS_CA_BUNDLE` for my company's certificate. That doesn't help reflex as it uses httpx & certifi (as fas as I can understand from `_init` method).
The CA bundle path for certifi can be found by `python -m certifi`.

So I ended up updating the bundle using `cat $REQUESTS_CA_BUNDLE >> $(python -m certifi)`.

I could initialise a project after that.

**Reply by Xinja-T** [reply]
could you please guide how you resolved this?
